patches/api/0347-Collision-API.patch

```diff
@@ -0,0 +1,62 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Wed, 6 Oct 2021 20:10:36 -0400
+Subject: [PATCH] Collision API
+
+
+diff --git a/src/main/java/org/bukkit/RegionAccessor.java b/src/main/java/org/bukkit/RegionAccessor.java
+index 35e407ec7bd8d5cd38069fe7f2cf3d9d3bc94cbe..addae0c86e47a9ee8c34619797e1a1faf8a4d9f8 100644
+--- a/src/main/java/org/bukkit/RegionAccessor.java
++++ b/src/main/java/org/bukkit/RegionAccessor.java
+@@ -355,4 +355,18 @@ public interface RegionAccessor {
+      */
+     @NotNull
+     public <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz, boolean randomizeData, @Nullable Consumer<T> function) throws IllegalArgumentException;
++
++    // Paper Start - Collision API
++
++    /**
++     *
++     * Checks for any collisions in the given boundingbox.
++     * This will check for any colliding hard entities / worldborder / blocks.
```

Comment:
What exactly are "hard entities" ? Might need clarification that this mainly wraps boats, shulkers and minecarts (as of right now).

Reply:
Gave some examples!